File: algo/LastStoneWeight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> q = new PriorityQueue<>((a, b) -> b-a);
        for (int stone : stones) {
            q.add(stone);
        }
        while (q.size() > 1) {
            q.add(Math.abs(q.poll()-q.poll()));
        }
        return q.poll();
    }
}

File: algo/LastStoneWeight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> q;
        for (int i = 0; i < stones.size(); i++) {
            q.push(stones[i]);
        }
        while (q.size() > 1) {
            int y = q.top();
            q.pop();
            int x = q.top();
            q.pop();
            q.push(y-x);
        }
        return q.top();
    }
};